src/herbivoresAndCarnivores.js

```diff
@@ -1,16 +1,60 @@
 'use strict';
 
 class Animal {
-  // write your code here
-}
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+};
 
 class Herbivore extends Animal {
-  // write your code here
-}
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    if(!this.hidden) {
+      this.hidden = true;
+    } else {
+      console.error('This animal is already in hiding.')
+    }
+  }
+};
 
 class Carnivore extends Animal {
-  // write your code here
-}
+  constructor(name, health = 100) {
```

Comment:
Im not sure, but carnivore doesnt need overriding constructor from parent. We don't need to add another property to carnivore. Take it with a grain of salt